frontend/src/core/l10n/components/AppLocalizationProvider.js

```diff
@@ -29,9 +30,21 @@ type InternalProps = {|
  */
 export class AppLocalizationProviderBase extends React.Component<InternalProps> {
     componentDidMount() {
-        // $FLOW_IGNORE: we count on the 'lang' attribute being set.
-        const locale = document.documentElement.lang;
-        this.props.dispatch(l10n.actions.get([locale]));
+        if (document.documentElement && document.documentElement.lang) {
+            // The user has chosen a specific locale, show that.
+            this.props.dispatch(
+                l10n.actions.get([ document.documentElement.lang ])
+            );
+        }
+        else {
+            // No specified locale, let's figure out the user's preferences.
+            const languages = negotiateLanguages(
+                navigator.languages,
+                ['fr', 'es', 'en-US'],  // available locales
```

Comment:
Is there an established way to put this in some kind of a config file? I'd recommend to also document that new languages need to be added here before they become available in the UI.

Reply:
I've filed https://bugzilla.mozilla.org/show_bug.cgi?id=1507811 to take care of this issue. I didn't want to spend more time on this right now. 

Note: your comments on how to solve this are very welcome in the bug! :-)

Reply:
Good point! Note that this will be taken care of (at least partly) in https://bugzilla.mozilla.org/show_bug.cgi?id=1507811.

Reply:
Good timing!